### PR TITLE
[3.4] Backport e2e tests for livez/readyz. 

### PR DESCRIPTION
**Pull request by siyuanfoundation**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Similar to https://github.com/etcd-io/etcd/pull/17083

### CODE REVIEW

File: tools/mod/install_all.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
```
Comment (fuweid) [inline tools/mod/install_all.sh:2]: suggest to add `set -euo pipefail` here

File: Makefile [hunk 2]
```diff
@@ -532,3 +532,9 @@ pull-docker-functional:
 	$(info GO_VERSION: $(GO_VERSION))
 	$(info ETCD_VERSION: $(ETCD_VERSION))
 	docker pull gcr.io/etcd-development/etcd-functional:go$(GO_VERSION)
+
+# Failpoints
+GOFAIL_VERSION = $(shell cd tools/mod && go list -m -f {{.Version}} go.etcd.io/gofail)
+.PHONY: install-gofail
+install-gofail:
+	go install go.etcd.io/gofail@${GOFAIL_VERSION}
```
Comment (fuweid) [inline Makefile:540]: please consider to remove last char

Comment (siyuanfoundation) [reply Makefile]: done

File: tools/mod/install_all.sh (hunk 1 above)
Comment (siyuanfoundation) [reply tools/mod/install_all.sh:2]: done

File: tests/e2e/etcd_process.go [hunk 3]
```diff
@@ -105,6 +125,15 @@ func (ep *etcdServerProcess) Start() error {
 	if ep.proc != nil {
 		panic("already started")
 	}
+	if ep.cfg.proxy != nil && ep.proxy == nil {
+		// ep.cfg.lg.Info("starting proxy...", zap.String("name", ep.cfg.name), zap.String("from", ep.cfg.proxy.From.String()), zap.String("to", ep.cfg.proxy.To.String()))
```
Comment (serathius) [inline tests/e2e/etcd_process.go:129]: Any reason to leave this here?

**Review by serathius — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -33,12 +33,12 @@ require (
...
-	golang.org/x/net v0.17.0
-	golang.org/x/sync v0.3.0
-	golang.org/x/sys v0.13.0
+	golang.org/x/crypto v0.15.0
+	golang.org/x/net v0.18.0
+	golang.org/x/sync v0.5.0
+	golang.org/x/sys v0.14.0
 	golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
-	google.golang.org/grpc v1.58.3
+	google.golang.org/grpc v1.59.0
```
Comment (ahrtr) [inline go.mod:41]: Usually we don't proactively bump dependency in a stable release unless there is any CVE or critical/major issue. Also we should do it in a separate PR if we decided to do it.

But I won't insist on it this time if there is no any objection from other maintainers and grpc-go expert/maintainer. @dfawley

Also interesting .... I see that the grpc-go's tag `v1.59.0` was created on Oct 17, 2023, but I do not see grpc-go team explicitly releases it, as I do not see the release in https://github.com/grpc/grpc-go/releases @dfawley is there any reason that you intentionally did not release grpc-go v1.59.0?

File: tests/e2e/etcd_process.go (hunk 3 above)
Comment (siyuanfoundation) [reply tests/e2e/etcd_process.go]: there is no log in 3.4 implementation of etcd process. Removed the line.

File: go.mod (hunk 4 above)
Comment (siyuanfoundation) [reply go.mod]: Removed the change since it is not necessary for this PR.

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Pull request merged**